### PR TITLE
CA-257178: Refine format string for messages

### PR DESCRIPTION
**Pull request by makunterry**
Signed-off-by: Kun Ma <kun.ma@citrix.com>

### CODE REVIEW

File: scripts/mail-alarm [hunk 1]
```diff
@@ -237,28 +238,29 @@ class NetworkUsageAlarmETG(EmailTextGenerator):
         pool_name = pool_name = get_pool_name()
```
Comment (mseri) [inline scripts/mail-alarm]: `pool_name = pool_name = get_pool_name()` should have been `pool_name = get_pool_name()`

File: scripts/mail-alarm [hunk 2]
```diff
@@ -277,27 +279,27 @@ class MemoryUsageAlarmETG(EmailTextGenerator):
...
```
Comment (mseri) [inline scripts/mail-alarm]: `pool_name` got duplicated here as well

Comment (makunterry) [inline scripts/mail-alarm]: Yes you are right. Thanks!

File: scripts/mail-alarm [hunk 3]
```diff
@@ -447,20 +445,19 @@ class WlbConsultationFailure(EmailTextGenerator):
...
-            '\n'
+            body_pattern = 'A workload balancing consultation for VM {name_label} failed.\n' \
+                           'The operation was completed using the default algorithm instead of a workload balancing recommendation.\n\n'
 
-        return body_pattern % \
-            (self.params['name_label'])
+        return body_pattern.format(name_label = self.params['name_label'])
             
 class WlbOptimizationAlert(EmailTextGenerator):
     def __init__(self, optimization_mode, severity, mail_language):
```
Comment (mseri) [inline scripts/mail-alarm]: another `pool_name` duplication on line 467: `self.pool_name = pool_name = get_pool_name()` -> `         self.pool_name = get_pool_name()`

**Comment by mseri** [comment]
Aside for the comments above, this looks good (and the direct style formatting is much more readable overall). It would be good in the future to use it for all the format strings in the script and also to replace all the `raise Exception, "some error"` (deprecated form) with `raise Exception("some error")` or, even better, `raise MailAlarmError("some error")` using a more meaningful exception class.

**Pull request merged**